types/ember-data/index.d.ts

```diff
@@ -919,10 +919,10 @@ export namespace DS {
      * it easy to create data bindings with the `PromiseArray` that will be
      * updated when the promise resolves.
      */
-    interface PromiseArray<T>
+    interface PromiseArray<T, ArrayType extends Ember.ArrayProxy<T>['content'] = Ember.Array<T>>
```

Comment:
If you default to Ember.ArrayProxy<T> instead, we can avoid needing to pass this in `class PromiseManyArray extends`, and the return types for `store.query` and `store.findAll`.
```suggestion
    interface PromiseArray<T, ArrayType extends Ember.ArrayProxy<T>['content'] = Ember.ArrayProxy<T>>
```
While still allowing it to be overridden with an array or EmberArrayLike.

Reply:
My thought is that, in general, we only can expect everything to be an `Ember.Array` and that I'd rather opt-in to the version that actually provides more as we are certain. For instance, this test for PromiseManyArray, just uses a regular `Ember.A` not a proxy: https://github.com/emberjs/data/blob/ff4f9111fcfa7dd9e39804ed17f5af27a4a01378/packages/-ember-data/tests/unit/promise-proxies-test.js

Reply:
Yeah, that makes sense. Even though all the usages of it _here_ are ArrayProxy, that's really a special case.